### PR TITLE
Fix issue with count(*) on >= 3 table joins

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -74,6 +74,9 @@ Changes
 Fixes
 =====
 
+ - Fixed an issue that caused and exception to be thrown when using
```
Comment (mfussenegger) [inline CHANGES.txt]: `and` -> `an` ?

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 2]
```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
...
+     *
+     *                   join
+     * outputs=t1[id]   /   \
+     *      |          /     \
+     *      +------> join    t3
+     *               / \
+     *              /   \
+     *             t1   t2
+     */
+    private static void buildJoinOutputs(Map<Set<QualifiedName>, Symbol> splitQuery,
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: maybe rename to somethingl ike `updateQSOutputs` or `extendQSOutputs` to at least indicate where the side effect happens.

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 3]
```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
         return join;
     }
 
+
+    /**
+     * Build outputs of joins not in the outer level of the tree
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: Maybe rephrase a bit. Something like "Extends the outputs of a querySpec to include symbols which are required by the next/upper join in the tree. These are queries that are not selected, but are for example used in a joinCondotion"

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 4]
```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
...
+     *               / \
+     *              /   \
+     *             t1   t2
+     */
+    private static void buildJoinOutputs(Map<Set<QualifiedName>, Symbol> splitQuery,
+                                         Iterator<QualifiedName> it,
+                                         final QualifiedName leftName,
+                                         final QualifiedName rightName,
+                                         QuerySpec newQuerySpec) {
+        if (it.hasNext() && newQuerySpec.outputs().isEmpty()) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: are you sure that this is only applicable if the outputs are empty? What if the query looks like `select count(t1.id) from t1, t2, t3 where t1.a = t2.b and ...`? (id is already selected, but the join condition uses another column)

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 5]
```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
...
+
+    /**
+     * Build outputs of joins not in the outer level of the tree
+     * based on the where queries that must be applied later on
+     * in case the current outputs are empty (case of count(*))
+     * e.g.:
+     * select count(*) from t1, t2, t3 where t1.a = t2.b and t2.b = t3.c
+     *
+     *                   join
+     * outputs=t1[id]   /   \
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: outputs would be t1.a in this case? (given the example above)

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java [hunk 6]
```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
...
+     */
+    private static void buildJoinOutputs(Map<Set<QualifiedName>, Symbol> splitQuery,
+                                         Iterator<QualifiedName> it,
+                                         final QualifiedName leftName,
+                                         final QualifiedName rightName,
+                                         QuerySpec newQuerySpec) {
+        if (it.hasNext() && newQuerySpec.outputs().isEmpty()) {
+            List<Symbol> fields = new ArrayList<>();
+            for (Map.Entry<Set<QualifiedName>, Symbol> entry : splitQuery.entrySet()) {
+                if (entry.getKey().contains(leftName) || entry.getKey().contains(rightName)) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: might want to assign `entry.getKey()` to a local variable (used twice).  Could do the same for entry.getValue() just for the sake of giving it a name (`joinCondition` probably?)

File: sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java (hunk 4 above)
Comment (matriv) [inline sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java]: You're right, adding new test and fixing the new method

**Comment by matriv** [comment]
@mfussenegger Plz check again. fixup pushed.
